src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1398,6 +1406,16 @@ void CodeGen::genSSEIntrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_SSE_X64_ConvertScalarToVector128Single:
```

Comment:
can't this just be grouped with `NI_SSE2_X64_ConvertScalarToVector128Double`, they look to have the same implementation.

Reply:
I remember that we discussed to merge `genSSEIntrinsic` and `genSSE2Intrinsic` (we already did for AVX and AVX2). Let me make it in the next PR.

Reply:
Was the intrinsic previously misnamed?  It looks like it used to be `NI_SSE2_ConvertScalrToVector128Single`: https://github.com/dotnet/coreclr/pull/21264/files#diff-847442c7efdf7a78e78711bea30fd4b2L1570

Reply:
We have both. I modified the table so that SSE2 `ConvertScalarToVector128Single` could be table-driven  https://github.com/dotnet/coreclr/pull/21264/files#diff-607364c1c16999989209b5ee9773829aR213
```csharp
//Sse
public static Vector128<float> ConvertScalarToVector128Single(Vector128<float> upper, int value) ;

// Sse.X64
public static Vector128<float> ConvertScalarToVector128Single(Vector128<float> upper, long value);

// Sse2
public static Vector128<float> ConvertScalarToVector128Single(Vector128<float> upper, Vector128<double> value);
```